src/renderer/helpers/walletHelper.ts

```diff
@@ -112,3 +112,17 @@ export const addressFromWalletAddress = ({ address }: Pick<WalletAddress, 'addre
 export const addressFromOptionalWalletAddress = (
   oWalletAddress: O.Option<Pick<WalletAddress, 'address'>>
 ): O.Option<Address> => FP.pipe(oWalletAddress, O.map(addressFromWalletAddress))
+
+export const getWalletByAddress = (
+  oWalletBalances: O.Option<NonEmptyWalletBalances>,
```

Comment:
Try to use `NonEmptyWalletBalances` w/o `O.Option` and change the logic in `matchedWalletType`